crawler/src/main/java/com/brandwatch/minibcr/crawler/service/reddit/auth/RedditAuthenticator.java

```diff
@@ -0,0 +1,46 @@
+package com.brandwatch.minibcr.crawler.service.reddit.auth;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+import com.brandwatch.minibcr.crawler.config.RestConfig;
+
+
+@Service
+public class RedditAuthenticator {
+
+    private static final String URL = "https://www.reddit.com/api/v1/access_token";
+
+    private final RestTemplate restTemplate;
+
+    public RedditAuthenticator(RestConfig config) {
+        this.restTemplate = config.restTemplate();
+    }
+
+    public String authenticate(RedditClient client) {
+        return getAuthToken(client);
+    }
+
+    private String getAuthToken(RedditClient client) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.setBasicAuth(client.getClientId(), client.getSecret());
+        headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+        headers.put("User-Agent", Collections.singletonList(client.getClientName()));
+        String body = "grant_type=client_credentials";
+        HttpEntity<String> request = new HttpEntity<>(body, headers);
+        ResponseEntity<Map<String, Object>> response = restTemplate.exchange(
+                URL, HttpMethod.POST, request, new ParameterizedTypeReference<Map<String, Object>>() {
+                });
+        return (String) Objects.requireNonNull(response.getBody()).get("access_token");
```

Comment:
Can this be handled differently that using the cast? For example but providing an appropriate return type in the request?